### PR TITLE
Replace babel in Jest tests

### PR DESCRIPTION
**Pull request by fregante**
Debugging #2000 and https://github.com/pixiebrix/pixiebrix-extension/issues/684 made me spend a little time in Jest tests and its config. I think we can simplify it a bit.

Pros:

- no more babel.config.js 🎉 
- drop 6 direct babel dependencies, add 2. Total number of deps gone from 3200 to 3031

### CODE REVIEW

File: jest.config.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "moduleFileExtensions": ["ts", "tsx", "js", "jsx", "yaml", "yml", "json"],
   "testPathIgnorePatterns": ["<rootDir>/src/vendors/", "<rootDir>/selenium/"],
   "transform": {
-    "^.+\\.[jt]sx?$": "babel-jest",
+    "^.+\\.[jt]sx?$": "@swc/jest",
```
Comment (fregante) [inline jest.config.json:8]: I went with `swc` instead of `esbuild` because it offers first-party jest support: https://github.com/swc-project/jest

**Review by fregante — COMMENTED** [review]
If merged, this should probably happen before other PRs because I rebuild the lockfile

**Comment by twschiller** [comment]
Get the following warnings about npm, but looks like it's working OK

```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@swc/jest@0.2.10',
npm WARN EBADENGINE   required: { npm: '^8.0.0' },
npm WARN EBADENGINE   current: { node: 'v16.6.0', npm: '7.20.3' }
npm WARN EBADENGINE }
```

**Comment by fregante** [comment]
Yeah it's fine, the packages only have 2/3 dependencies in total, nothing crazy that requires npm 8. It's all good

**Review by twschiller — APPROVED** [review]

**Comment by fregante** [comment]
Fixed https://github.com/swc-project/jest/pull/50

**Pull request merged**